src/Microsoft.ML.Tokenizers/Utils/Vec.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.ML.Tokenizers
+{
+    internal struct Vec<T>
```

Comment:
Is the reason you aren't using a list to get this to be a struct?

Reply:
Vec exists for a perf reason in some scenarios. 
- Vec doesn't allocate any memory by default when start using it. As it is a struct not allocating any object and not even allocating any underlying buffer till it gets used. 
- The more important reason is it allows modifying the item inside the list without the need for copying. Here is an example:

```C#
            List<MyStruct> list = new();
            list.Add(new MyStruct() { X = 10, Y = 20 });
            list[0].X = 30; // <<<<<<<<<< You cannot do that with the list, but you can do it with the Vec.
```